src/openscm_runner/adapters/fair2/_run_fair.py

```diff
@@ -0,0 +1,366 @@
+"""
+Module for running FaIR
+"""
+import logging
+
+import numpy as np
+from fair.constants.general import EARTH_RADIUS, SECONDS_PER_YEAR
+import fair.version_two.emissions_driven as fair_scm
+from scmdata import ScmRun, run_append
+
+LOGGER = logging.getLogger(__name__)
+toa_to_joule = 4 * np.pi * EARTH_RADIUS ** 2 * SECONDS_PER_YEAR
+
+
+def run_fair(cfgs, output_vars):  # pylint: disable=R0914
+    """
+    Run FaIR
+
+    Parameters
+    ----------
+    cfgs : list[dict]
+        List of configurations with which to run FaIR
+
+    output_vars : list[str]
+        Variables to output
+
+    Returns
+    -------
+    :obj:`ScmRun`
+        :obj:`ScmRun` instance with all results.
+    """
+    res = []
+
+    for cfg in cfgs:
+        scenario = cfg.pop("scenario")
+        model = cfg.pop("model")
+        run_id = cfg.pop("run_id")
+        factors = {}
+        factors["gmst"] = cfg.pop("gmst_factor")
+        factors["ohu"] = cfg.pop("ohu_factor")
+        data, unit, nt = _process_output(fair_scm(**cfg), output_vars, factors)
+
+        data_scmrun = []
+        variables = []
+        units = []
+        for key, variable in data.items():
+            variables.append(key)
+            data_scmrun.append(variable)
+            units.append(unit[key])
+
+        tempres = ScmRun(
+            np.vstack(data_scmrun).T,
```

Comment:
You can pass a dataframe to `ScmRun` which makes things way quicker, quicker again if you include `copy=False`